MineSweeper.py

```diff
@@ -0,0 +1,245 @@
+
+  marked_mines = 0
+
+  for row in board:
+    for tile in row:
+      tile.draw(panel)
+      
+      if(tile.marked_mine):
+        marked_mines += 1
+
+  mines = NUMBER_MINES - marked_mines
+
+  bottom_text = 'Mines left: ' + repr(mines)
+  text_fill = 'black'
+
+  panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 20, text  = bottom_text, fill = text_fill)
+
+
+# assign mines to random locations across the board. 
+def assign_mines(board):
+
+  mines_to_assign = NUMBER_MINES
+
+  while(mines_to_assign > 0):
+    x_index = random.randint(0, len(board)-1)
+    y_index = random.randint(0, len(board)-1)
+
+
+    if(board[x_index][y_index].is_mine == False):
+      board[x_index][y_index].is_mine = True
+      mines_to_assign -= 1
+
+      for i in range(-1, 2):
+        for j in range(-1, 2):
+          try:
+            if((x_index + i >= 0) and (y_index + j >= 0)):
+              board[x_index + i][y_index + j].value += 1
+          except IndexError:
+            pass
+
+# When the user left-clicks, gather they x, y data and reveal the tile 
+# that surrounds the coordinates.
+def left_click_event(event):
+ 
+  mouse_click = []
+
+  mouse_click.append(event.x)
+  mouse_click.append(event.y)
+
+  box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+  box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+  try:
+    board[box_index_x][box_index_y].show_self(box_index_x, box_index_y)
+    
+    if(board[box_index_x][box_index_y].is_mine):
+      game_over()
+
+  except IndexError:
+    pass
+  
+  draw_board(False) 
+
+# Right click event. Marks a square as a mine
+def flag_square(event):
+
+  mouse_click = []
+
+  mouse_click.append(event.x)
+  mouse_click.append(event.y)
+
+  box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+  box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+
+  try:
+    board[box_index_x][box_index_y].marked_mine = True
+
+
+    draw_board(False)
+
+  except IndexError:
+    pass
+
+    
+def game_over():
+
+  master.unbind("<Button-1>")
+  master.unbind("<Button-2>")
+
+  #draw_board(True)
+  panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 40, text  = 'Game Over', fill = 'red') 
+
+  b = Button(master, text = 'Play Again?', command = main)
+  b.pack()
+
+def main():
+
+  global mines, master, panel, board
+
+  try:
+    if(master):
+      master.delete(b)
+      master.quit()
+      panel.destroy()
+
+  except NameError:
+    pass
+
+  board = []
+
+  master = tk.Tk()
+
+  panel = Canvas(master, width = BOARD_WIDTH, height = BOARD_HEIGHT + 100, bg = 'grey')
+  panel.pack()
+
+  mines = NUMBER_MINES
+
+  boxes_x = BOARD_WIDTH / BOX_SIZE - 1
+  boxes_y = BOARD_HEIGHT / BOX_SIZE - 1
+
+  row = []
+
+  for i in range(boxes_y):
+    for j in range(boxes_x):
+      new_tile = Tile(i*BOX_SIZE, j*BOX_SIZE, BOX_SIZE)
+      row.append(new_tile)
+
+    board.append(row)
+    row = []
+
+  assign_mines(board)  
+
+  draw_board(False)
+
+  master.bind("<Button-1>", left_click_event)
+  master.bind("<Button-2>", flag_square)
```

Comment:
I noticed in your event handlers that one of the handlers is named after the button used to trigger it (`left_click_event`) and the other is named after what it does (`flag_square`). Can you change them so they are consistent with each other?